src/Tes.Runner/Executor.cs

```diff
@@ -4,42 +4,80 @@
 using System.Diagnostics;
 using Microsoft.Extensions.Logging;
 using Tes.Runner.Docker;
+using Tes.Runner.Events;
 using Tes.Runner.Models;
 using Tes.Runner.Storage;
 using Tes.Runner.Transfer;
 
 namespace Tes.Runner
 {
-    public class Executor
+    public class Executor : IAsyncDisposable
     {
         public const long ZeroBytesTransferred = 0;
+        public const long DefaultErrorExitCode = -1;
```

Comment:
In Linux, exit codes can only be 0-255, so it would be better to use "1" as the error exit code here